contracts/src/xchain.ts

```diff
@@ -385,15 +385,16 @@ export function xchainRPC(chainId:number)
     return new JsonRpcProvider(rpc_url);
 }
 
-export async function tokenDetailsFromProvider(addr:string, provider:JsonRpcProvider) : Promise<TokenInfo>
+const ERC20Abi = [
+  "function name() public view returns (string)",
+  "function symbol() public view returns (string)",
+  "function decimals() public view returns (uint8)",
+  "function totalSupply() public view returns (uint256)",
+];
+
+export async function ERC20TokenDetailsFromProvider(addr:string, provider:JsonRpcProvider) : Promise<TokenInfo>
```

Comment:
I am somewhat conflicted by the name. It should be lowercase since it's a function, but `erc20Whatever` also looks a bit weird. Opinions?

Reply:
We should be able to support [ERC-721](https://eips.ethereum.org/EIPS/eip-721) and [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155) as well as ERC-20.

They all support the `balanceOf` function.

However, `name`, `symbol`, `decimals` and `totalSupply` aren't consistent across them.